packages/http-specs/specs/payload/pageable/main.tsp

```diff
@@ -248,5 +293,122 @@ namespace ServerDrivenPagination {
 
       @continuationToken @header nextToken?: string;
     };
+
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination with nested response structure. Continuation token is passed in the request query and nested within response body.
+      
+      Two requests need to be tested.
+      
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-nested-response-body?bar=bar
+      
+      Expected request header:
+      foo=foo
+      
+      Expected response body:
+      ```json
+      { "nestedItems": {
+          "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ]
+        },
+        "nestedNext": {
+          "nextToken": "page2"
+        }
+      }
+      ```
+      
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-nested-response-body?bar=bar&token=page2
+      
+      Expected request header:
+      foo=foo
+      
+      Expected response body:
+      ```json
+      { "nestedItems": {
+          "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+          ]
+        }
+      }
+      ```
+      """)
+    @route("/request-query-nested-response-body")
+    @list
+    op requestQueryNestedResponseBody(
+      @continuationToken @query token?: string,
+      ...HeaderAndQuery,
+    ): {
+      nestedItems: {
+        @pageItems
+        pets: Pet[];
+      };
+      nestedNext?: {
+        @continuationToken nextToken?: string;
+      };
+    };
```

Comment:
Why do we not use a named model?

Reply:
@weidongxu-microsoft Just for convenience. If Java needs, I could define a named model.